src/BitStringAddresses/fockaddress.jl

```diff
@@ -0,0 +1,188 @@
+"""
+    AbstractFockAddress
+
+Supertype representing a Fock state.
+"""
+abstract type AbstractFockAddress end
```

Comment:
Would it make sense to have the number of particle and modes as type parameters?
```suggestion
abstract type AbstractFockAddress{M,N} end
```
Then `num_particles` and `num_modes` would not have to be defined on the concrete types.
The logic being that a Fock state has well defined particle number (for us it also has a fixed and finite number of modes out of necessity).